@payloadcms/live-preview creates dependencies to Payload

### Link to reproduction

_No response_

### Describe the Bug

The library itself uses a function from Payload [here](https://github.com/payloadcms/payload/blob/191c13a40946e6bcba07b97d7784f88e87449856/packages/live-preview/src/mergeData.ts#L1) and uses Payload as `peerDependency`. This may work great when you're using Next and Payload together in the same project, but in docs says `No matter what front-end framework you are using, you can build your own hook using the same underlying tooling that Payload provides.`.

**Issue 1**: the library installs `payload` + its dependencies => a lot of dependencies for a simple library
**Issue 2**: Using `yarn`/`pnpm` can cause errors (see **To Reproduce**)

### To Reproduce

Create an Angular application and try to add `@payloadcms/live-preview`.  You will receive errors like:

```
Error: node_modules/.pnpm/payload@2.1.1_typescript@5.2.2_webpack@5.89.0/node_modules/payload/dist/config/types.d.ts:2:58 - error TS7016: Could not find a declaration file for module 'express-fileupload'. 'C:/Projects/gvhdev.com/gvhdev.com.angular/node_modules/.pnpm/express-fileupload@1.4.0/node_modules/express-fileupload/lib/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/express-fileupload` if it exists or add a new declaration (.d.ts) file containing `declare module 'express-fileupload';`

2 import type { Options as ExpressFileUploadOptions } from 'express-fileupload';
```

### Payload Version

-

### Adapters and Plugins

_No response_

--- a/packages/live-preview-react/package.json
+++ b/packages/live-preview-react/package.json
@@ -25,7 +25,6 @@
     "payload": "workspace:*"
   },
   "peerDependencies": {
-    "payload": "^2.0.0",
     "react": "^16.8.0 || ^17.0.0 || ^18.0.0"
   },
   "exports": {

--- a/packages/live-preview/package.json
+++ b/packages/live-preview/package.json
@@ -20,9 +20,6 @@
     "@payloadcms/eslint-config": "workspace:*",
     "payload": "workspace:*"
   },
-  "peerDependencies": {
-    "payload": "^2.0.0"
-  },
   "exports": {
     ".": {
       "default": "./src/index.ts",

